### PR TITLE
Allow false as check_box unchecked value

### PR DESCRIPTION
**Pull request by pedrosmmoreira**
### Summary

When using boolean false as value for `check_box` unchecked value, the
hidden input for unchecked value is not generated.

Presently, the following example:

```ruby
<%= form_with :model do |form_builder| %>
  <%= form_builder.check_box :attribute, {}, true, false %>
<% end %>
```

will only output the input for true, ie:

```html
<input id="model_attribute name="model[attribute]" type="checkbox" value="true" />
```

instead of the expected behaviour:

```html
<input name="model[attribute]" type="hidden" value="false" />
<input id="model_attribute name="model[attribute]" type="checkbox" value="true" />
```

This is somewhat surprising behaviour, in particular since a different order of values will work correctly, ie:

```ruby
<%= form_with :model do |form_builder| %>
  <%= form_builder.check_box :attribute, {}, false, true %>
<% end %>
```

as well as using strings:

```ruby
<%= form_with :model do |form_builder| %>
  <%= form_builder.check_box :attribute, {}, "false", "true" %>
<% end %>
```

This is due to a ternary operator being used to determine if an unchecked value is present. On receiving a boolean false, an empty string is returned.

This PR changes the behaviour to cast the unchecked value to a string and calling `.present?` on it, so as to allow the expected behaviour to occur.

### CODE REVIEW

**Comment by p8** [comment]
This seems like a correct fix. I don't see a reason why we would not render the unchecked value when it is false.
It looks like this case was overlooked before as there was no test case.

File: actionview/lib/action_view/helpers/tags/check_box.rb [hunk 1]
```diff
@@ -57,7 +57,7 @@ def checked?(value)
           end
 
           def hidden_field_for_checkbox(options)
-            @unchecked_value ? tag("input", options.slice("name", "disabled", "form").merge!("type" => "hidden", "value" => @unchecked_value)) : "".html_safe
+            @unchecked_value.to_s.present? ? tag("input", options.slice("name", "disabled", "form").merge!("type" => "hidden", "value" => @unchecked_value)) : "".html_safe
```
Comment (rafaelfranca) [inline actionview/lib/action_view/helpers/tags/check_box.rb:60]: `present?` here will make `""` not be a valid unchecked value anymore. I think we should just do
```suggestion
            !@unchecked_value.nil? ? tag("input", options.slice("name", "disabled", "form").merge!("type" => "hidden", "value" => @unchecked_value)) : "".html_safe
```

A test is needed for the `""` as unchecked value.

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

File: actionview/lib/action_view/helpers/tags/check_box.rb (hunk 1 above)
Comment (pedrosmmoreira) [reply actionview/lib/action_view/helpers/tags/check_box.rb:60]: Thanks, @rafaelfranca ! Will do

**Comment by pedrosmmoreira** [comment]
Hi @rafaelfranca, I've updated the PR as per the comments - apologies on the long turnaround time.

**Pull request closed without merging**